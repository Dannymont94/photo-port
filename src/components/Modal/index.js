import React from 'react';

function Modal({ currentPhoto, toggleModal }) {
  const { name, category, description, index } = currentPhoto;
  return (
    <div className="modalBackdrop">
      <div className="modalContainer">
        <div className="flex-row space-between">
          <h3 className="modalTitle">{name}</h3>
          <button onClick={toggleModal} type="button">
            Close
          </button>
        </div>
        <img src={require(`../../assets/large/${category}/${index}.jpg`)} alt={description} />
        <p>{description}</p>
      </div>
    </div>
  );
}

export default Modal;