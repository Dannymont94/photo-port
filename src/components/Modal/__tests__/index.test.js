import React from 'react';
import { render, cleanup, fireEvent } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import Modal from '..';

const mockToggleModal = jest.fn();

const currentPhoto = {
  name: 'Grocery aisle',
  category: 'commercial',
  description: 'Lorem ipsum dolor sit amet, consectetur adipiscing elit. Nunc ultricie',
  index: 1
};

afterEach(cleanup);

describe('Modal component', () => {
  it('renders', () => {
    render(<Modal currentPhoto={currentPhoto} toggleModal={mockToggleModal}/>);
  });

  it('matches snapshot', () => {
    const { asFragment } = render(<Modal currentPhoto={currentPhoto} toggleModal={mockToggleModal} />);

    expect(asFragment()).toMatchSnapshot();
  });
});

describe('Click event', () => {
  it('calls toggleModal', () => {
    const { getByText } = render(<Modal toggleModal={mockToggleModal} currentPhoto={currentPhoto} />);

    fireEvent.click(getByText('Close'));

    expect(mockToggleModal).toHaveBeenCalledTimes(1);
  });
});