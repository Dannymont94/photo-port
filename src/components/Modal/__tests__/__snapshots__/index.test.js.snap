// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Modal component matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="modalBackdrop"
  >
    <div
      class="modalContainer"
    >
      <div
        class="flex-row space-between"
      >
        <h3
          class="modalTitle"
        >
          Grocery aisle
        </h3>
        <button
          type="button"
        >
          Close
        </button>
      </div>
      <img
        alt="Lorem ipsum dolor sit amet, consectetur adipiscing elit. Nunc ultricie"
        src="1.jpg"
      />
      <p>
        Lorem ipsum dolor sit amet, consectetur adipiscing elit. Nunc ultricie
      </p>
    </div>
  </div>
</DocumentFragment>
`;
